factory/src/main/kotlin/io/qalipsis/core/factory/meters/DistributionSummaryImpl.kt

```diff
@@ -19,21 +19,54 @@
 
 package io.qalipsis.core.factory.meters
 
+import io.aerisconsulting.catadioptre.KTestable
+import io.qalipsis.api.meters.DistributionMeasurementMetric
 import io.qalipsis.api.meters.DistributionSummary
+import io.qalipsis.api.meters.Measurement
+import io.qalipsis.api.meters.MeasurementMetric
 import io.qalipsis.api.meters.Meter
+import io.qalipsis.api.meters.MeterSnapshot
+import io.qalipsis.api.meters.Statistic
 import io.qalipsis.api.report.ReportMessageSeverity
 import io.qalipsis.core.reporter.MeterReporter
+import java.time.Instant
+import java.util.TreeSet
 import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.DoubleAdder
+import java.util.concurrent.atomic.LongAdder
+import kotlin.Comparator
+import kotlin.math.ceil
 
+/**
+ * Implementation of distribution summary to determine the statistical distribution of events.
+ *
+ * @author Francisca Eze
+ */
 internal class DistributionSummaryImpl(
-    val micrometer: io.micrometer.core.instrument.DistributionSummary,
     override val id: Meter.Id,
-    private val meterReporter: MeterReporter
-) : DistributionSummary, Meter.ReportingConfiguration<DistributionSummary>,
-    io.micrometer.core.instrument.DistributionSummary by micrometer {
+    private val meterReporter: MeterReporter,
+    private val percentiles: Collection<Double>,
+    private val histogramCount: Collection<Long>,
+) : DistributionSummary, Meter.ReportingConfiguration<DistributionSummary> {
 
     private var reportingConfigured = AtomicBoolean()
 
+    @KTestable
+    private val observationCount = LongAdder()
+
+    @KTestable
+    private val total = DoubleAdder()
+
+    @KTestable
+    private val observationBucket = mutableListOf<Double>()
```

Comment:
I searched a better solution and found out that we should use a T-Digest: https://github.com/tdunning/t-digest.